### ISSUE TITLE
add touchTolerance: clickTolerance for touch events

### ISSUE THREAD

**Issue by hyperknot** [mapbox/mapbox-gl-js#10061]
## Motivation

A larger clickTolerance (like 20 pixels) makes sense for mobile/touch devices. Yet setting the same setting makes mouse usage really unpleasant, a simple panning starts with a "stutter", a slight freeze + jump.

## Design Alternatives

Since clickTolerance is implemented in many levels already (MouseRotateWrapper, BoxZoomHandler, MouseHandler, TouchPanHandler) it'd be great to set it independently for click and touch events. For example, I'd like to set it to 20 pixels for touch events, while leaving it on 3 pixels for mouse events.

Alternatively, I'd be happy to use a workaround if there exists one. My only idea so far is to try to figure out device type using [detect-it](https://github.com/rafgraph/detect-it), like:

```
clickTolerance: detectIt.primaryInput === 'mouse' ? 3 : 20
```

The problem with this is that it is very much sub-optimal as it sets one global value for the whole device, with no idea how hybrid devices are used. Since Mapbox GL JS handles each event differently, it'd be great to specify this at init time, like:

```
clickTolerance: 3,
touchTolerance: 20,
```

**Comment by mourner** [comment]
Yeah, this sounds like a good idea, especially given the increasing share of hybrid touch/mouse devices such as Surface. This is also consistent with Leaflet's different options of `clickTolerance` vs `tapTolerance`. We could have different values set by default for better overall experience.
